devicetypes/smartthings/zigbee-ceiling-fan-light.src/itm-fan-child.groovy

```diff
@@ -0,0 +1,97 @@
+/*
+ *  Copyright 2017 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ *  ZigBee Ceiling Fan Light
+ *
+ *  Author: SAMSUNG LED
+ *  Date: 2021-06-30
+ */
+
+import physicalgraph.zigbee.zcl.DataType
+import groovy.json.JsonOutput
+
+metadata {
+	definition(name: "ITM Fan Child", namespace: "SAMSUNG LED", author: "SAMSUNG LED", ocfDeviceType: "oic.d.fan") {
+		capability "Actuator"
+        	capability "Configuration"
+        	capability "Refresh"
+        	capability "Switch"
+		/* Capability "Switch Level" is used to control fan speed for platforms don't support capability "Fan speed"
+		 * when you connect other platforms via SmartThings cloud to cloud connection. */
+		capability "Switch Level"
+        	capability "Fan Speed"
+    }    
+}
+
+def off() {
+	setFanSpeed(0)
+}
+
+def on() {
+	setFanSpeed(1)
+}
+
+def setLevel(value){
+	if (value <= 1) {
+		setFanSpeed(0)
+        } else if (value <= 25) {
+		setFanSpeed(1)
+        } else if (value <= 50) {
+		setFanSpeed(2)
+        } else if (value <= 75) {
+		setFanSpeed(3)
+	} else if (value <= 100) {
+		setFanSpeed(4)
+       	}	
+}
+
+def setFanSpeed(speed) {
+	log.debug "child setFanSpeed $speed"	
+	if (speed as Integer == 0) {
+        	sendEvent(name: "switch", value: "off", displayed: true, isStateChange: true)
+    	} else {
+        	sendEvent(name: "switch", value: "on", displayed: true, isStateChange: true)
+    	}	
```

Comment:
I'd prefer if those events were created as a response of the command, not automatically when command is triggered.

Reply:
@RaihaPark what about this?

Reply:
@RaihaPark I'd really prefer if you'd address it. At least explain why do you think it's not a good idea to change it.

Reply:
@RaihaPark I've guess you've missed this conversation.

Reply:
Oh, I'm sorry. I missed it. But sometimes when I change the swithlevel, switch does not change to "on" or "off" in other platforms. So I put this code in.

Reply:
Maybe it's caused by too low delay here? https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/68445/files#diff-0983345aeab4215b6a7f7c49ebfe0e649324ac2d64018e6c8bffe0c9b24692b6R123